docs/Server/API/V1/relay.md

```diff
@@ -0,0 +1,143 @@
+# `POST /api/v1/relay/<uri>`
+
+This API creates a dataset resource by reading data from a Relay server in two
+steps. The provided URI represents a "manifest file" stored on the Relay server,
+a JSON file (`application/json` MIME format) defining the original tarball name,
+the MD5 hash value, optional metadata key/value pairs for the dataset, and a
+URI from which the Pbench Server expects to `GET` an `application/octet-stream`
+of a file created by `tar` and compressed with the `xz` program.
+
+Primarily this is expected to be a native Pbench Agent tarball with a specific
+structure; however with the `server.archiveonly` metadata key the Pbench Server
+can be used to archive and manage metadata for any tarball passed through a
+Relay server.
+
+## URI parameters
+
+`<uri>` string \
+The Relay server URI of the tarball's manifest `application/json` file. This
+must provide the following JSON keys:
+
+For example,
+
+```json
+{
+    "uri": "https://relay.example.com/52adfdd3dbf2a87ed6c1c41a1ce278290064b0455f585149b3dadbe5a0b62f44",
+    "md5": "22a4bc5748b920c6ce271eb68f08d91c".
+    "name": "fio_rw_2018.02.01T22.40.57.tar.xz".
+    "access": "private",
+    "metadata": ["server.origin:myrelay", "global.agent:cloud1"]
+}
+```
+
+## Manifest file keys
+
+`access`: [ `private` | `public` ] \
+The desired initial access scope of the dataset. Select `public` to make the
+dataset accessible to all clients, or `private` to make the dataset accessible
+only to the owner. The default access scope if the key is omitted from the
+manifest is `private`.
+
+For example, `"access": "public"`
+
+`md5`: tarball MD5 hash \
+The MD5 hash of the compressed tarball file. This must match the actual tarball
+octet stream specified by the manifest `uri` key.
+
+`metadata`: [metadata key/value strings] \
+A set of desired Pbench Server metadata key values to be assigned to the new
+dataset. You can set the initial resource name (`dataset.name`), for example, as
+well as assigning any keys in the `global` and `user` namespaces. See
+[metadata](../metadata.md) for more information.
+
+In particular the client can set any of:
+* `dataset.name`: [default dataset name](../metadata.md#datasetname)
+* `server.origin`: [dataset origin](../metadata.md#serverorigin)
+* `server.archiveonly`: [suppress indexing](../metadata.md#serverarchiveonly)
+* `server.deletion`: [default dataset expiration time](../metadata.md#serverdeletion).
+
+For example, `?metadata=server.archiveonly:true,global.project:oidc`
+
+`name`: The original tarball file name \
+The string value must represent a legal filename with the compound type of
+`.tar.xz` representing a `tar` archive compressed with the `xz` program.
```

Comment:
Is this suffix still required if `server.archiveonly` is set?

Reply:
Yeah, it is: the server has always required `.tar.xz` in the specified filename.

We'll probably need to change that going forward, either to generally loosen it or at least check against a list; but nothing I've done so far attacks that specific issue, including `server.archiveonly`.